test/JDBC/expected/BABEL-2218.out

```diff
@@ -42,10 +45,79 @@ DECLARE @ret INT;
 SET @ret = f2218();
 SELECT @ret;
 DROP FUNCTION f2218;
-DROP TABLE t2218;
 GO
 ~~START~~
 int
-2218
+2219
 ~~END~~
 
+
+-- Throw error if cursor for select doesn't have a destination(INTO @variable) inside a function BABEL-4586
+CREATE FUNCTION f_getval()RETURNS INTEGER
+AS
+BEGIN
+DECLARE temp_cursor CURSOR FOR SELECT a FROM t2218
+  OPEN temp_cursor
+  FETCH NEXT FROM temp_cursor
+  CLOSE temp_cursor
+  RETURN 1
+  END
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: SELECT statements included within a function cannot return data to a client.)~~
+
+
+-- cursor for select work if the destination(INTO @variable) is provided inside a function BABEL-4586
+CREATE FUNCTION f_getval() RETURNS INTEGER
+AS
+BEGIN
+DECLARE @my_var int
+DECLARE temp_cursor CURSOR FOR SELECT a FROM t2218
+  OPEN temp_cursor
+  FETCH NEXT FROM temp_cursor INTO @my_var
+  CLOSE temp_cursor
+  RETURN @my_var
+  END
+go
+
+DROP FUNCTION f_getval;
+GO
+
+-- cursor for select work with multiple fetch if the destination(INTO @variable) is provided inside a function BABEL-4586
+CREATE FUNCTION f_getval() RETURNS INTEGER
+AS
+BEGIN
+DECLARE @my_var int
+DECLARE temp_cursor CURSOR FOR SELECT a FROM t2218
+  OPEN temp_cursor
+  FETCH NEXT FROM temp_cursor INTO @my_var
+  FETCH NEXT FROM temp_cursor INTO @my_var
+  CLOSE temp_cursor
+  RETURN @my_var
+  END
+go
+
+DROP FUNCTION f_getval;
+GO
+
+-- cursor for select should throw error even if one fetch tries to return results to client
+CREATE FUNCTION f_getval() RETURNS INTEGER
+AS
+BEGIN
+DECLARE @my_var int
+DECLARE temp_cursor CURSOR FOR SELECT a FROM t2218
+  OPEN temp_cursor
+  FETCH NEXT FROM temp_cursor INTO @my_var
+  FETCH NEXT FROM temp_cursor
+  CLOSE temp_cursor
+  RETURN @my_var
+  END
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: SELECT statements included within a function cannot return data to a client.)~~
+
```

Comment:
We can also add test cases for FETCH CURSOR stmts inside stored procedures & triggers.

Reply:
Added